src/main/java/com/epam/ta/reportportal/core/item/identity/TestItemUniqueIdGenerator.java

```diff
@@ -84,10 +109,31 @@ private String prepareForEncoding(TestItem testItem, List<Long> parentIds, Launc
   }
 
   private List<String> getPathNames(List<Long> parentIds) {
+    List<TestItem> testItems = getTestItems(parentIds);
     return testItemRepository.findAllById(parentIds)
         .stream()
         .sorted(Comparator.comparingLong(TestItem::getItemId))
         .map(TestItem::getName)
         .collect(Collectors.toList());
   }
+
+  private List<TestItem> getTestItems(List<Long> parentIds) {
+    List<TestItem> testItems = new ArrayList<>();
+    for(Long id: parentIds) {
+      TestItem testItem = itemsCache.getIfPresent(id);
+      System.out.println("TestItem id " + id);
+      if (testItem == null) {
+        dbQueries++;
+        System.out.println("TestItemUniqueIdGenerator db queries: " + dbQueries);
+        testItem = testItemRepository.getById(id);
```

Comment:
Usually such calling the database in a loop is a place for performance improvement. Perhaps these calls could be executed in paralell or by single request. Also `parentIds` variable could be of `Set<>` type in order to avoid duplicates